CMakeLists.txt

```diff
@@ -100,6 +100,15 @@ check_include_file("unistd.h" HAVE_UNISTD_H)
 check_symbol_exists(snprintf "stdio.h" HAVE_SNPRINTF)
 check_symbol_exists(strcasecmp "strings.h" HAVE_STRCASECMP)
 
+find_package(OpenSSL)
```

Comment:
My preference is that we can default to picohash unless users want to opt into openssl. At the least, we probably want a way to avoid using OpenSSL explicitly from the command line.

Reply:
Ok great, I added a `-DUSE_INTERNAL_SHA256=ON` option that makes this explicit.